sherpa/astro/io/crates_backend.py

```diff
@@ -166,51 +182,55 @@ def _try_key(crate, name, dtype=str):
     return dtype(value)
 
 
-def _get_meta_data(crate):
-    """
-    checked for new crates
+def _get_meta_data(crate: CrateType) -> dict[str, KeyType]:
+    """Retrieve the header information from the crate.
+
+    This loses the "specialized" keywords like HISTORY and COMMENT.
+
     """
+
     meta = {}
-    names = crate.get_keynames()
-    if names is not None:
-        for name in names:
-            val = crate.get_key(name).value
+    for name in crate.get_keynames():
+        val = crate.get_key(name).value
 
-            # empty numpy strings are not recognized by load pickle!
-            if type(val) is numpy.str_ and val == '':
-                val = ''
+        # empty numpy strings are not recognized by load pickle!
+        if isinstance(val, np.str_) and val == '':
+            val = ''
```

Comment:
This is old code and I've been wary of changing it too much. It looks like it's trying to convert from a NumPy string to a "normal" one, but why I don't know (I know there have been historical issues with passing data around, for a variety of reasons). So this is one area I decided not to touch (I am running tests with a crates backend but I'm slightly careful as we don't have a CI run with it). 

I can look again at this code, as it's been a while since I last worked on it.

Reply:
This is very-much a "I'm sure there was a reason for this, but I don't know what". As we don't have a test that exercises this I am a little reluctant to change it at this time (which is somewhat funny giving the sweeping changes I'm making to the rest of the I/O code here...).